iothub/device/src/Message.cs

```diff
@@ -107,6 +107,32 @@ internal Message(Stream stream, bool ownStream)
             this.ownsBodyStream = ownStream;
         }
 
+        /// <summary>
+        /// Clones an existing <see cref="Message"/> instance and sets content body defined by <paramref name="byteArray"/> on it.
+        /// </summary>
+        /// <param name="byteArray">Message content to be set after clone.</param>
+        /// <returns>A new instance of <see cref="Message"/> with body content defined by <paramref name="byteArray"/>,
+        /// and user/system properties of the cloned <see cref="Message"/> instance.
+        /// </returns>
+        /// <remarks>user should treat the input byte array as immutable when
+        /// sending the message.</remarks>
+        public Message CloneWithBody(byte[] byteArray)
```

Comment:
Will VS let you use the `in` keyword for this parameter?

Reply:
yes, it works (after I updated VS)